akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -922,6 +922,35 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       }
     }
 
+    "enforce settings" should {
+
+      "reject new substreams when exceeding SETTINGS_MAX_CONCURRENT_STREAMS" in new TestSetup with RequestResponseProbes with Http2FrameHpackSupport {
```

Comment:
I think we should also add a test about what happens if we skip stream IDs. In that case, the skipped streams should be automatically be set to `Closed` state.